dask/dataframe/core.py

```diff
@@ -727,38 +736,41 @@ def apply(func, columns=None):
                               self.frame, self.index)
 
     def sum(self):
-        chunk = lambda df, index: df.groupby(index)[self.key].sum()
-        agg = lambda df: df.groupby(level=0).sum()
+        chunk = lambda df, index: df.groupby(index).sum()
+        agg = lambda df: df.groupby(self.index).sum()
         return aca([self.frame, self.index],
                    chunk=chunk, aggregate=agg, columns=[self.key])
 
     def min(self):
-        chunk = lambda df, index: df.groupby(index)[self.key].min()
-        agg = lambda df: df.groupby(level=0).min()
+        chunk = lambda df, index: df.groupby(index).min()
+        agg = lambda df: df.groupby(self.index).min()
```

Comment:
Why did you end up needing this.  I'm running into an issue with `groupby(...).apply` that has me confused.